File: app/src/main/java/com/dimatest/movieapp/utils/DataBindingAdapter.java
package com.dimatest.movieapp.utils;

import android.widget.ImageView;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;

public class DataBindingAdapter {

    @BindingAdapter("loadPoster")
    public static void loadPoster(ImageView imageView, String url) {
        Glide.with(imageView.getContext())
                .load(url)
                .transform(new CenterCrop(), new RoundedCorners(24))
                .into(imageView);
    }

    @BindingAdapter("loadPosterWithBlur")
    public static void loadPosterWithBlur(ImageView imageView, String url) {
        Glide.with(imageView.getContext())
                .load(url)
                .transform(new BlurTransformation(imageView.getContext()))
                .into(imageView);
    }

    @BindingAdapter("rating")
    public static void setRating(TextView textView, Double rating) {
        textView.setText(String.valueOf(rating));
    }
}

File: app/src/main/java/com/dimatest/movieapp/utils/DataBindingAdapter.kt
package com.dimatest.movieapp.utils

import android.widget.ImageView
import android.widget.TextView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners

@BindingAdapter("loadPoster")
fun loadPoster(imageView: ImageView, url: String?) {
    Glide.with(imageView.context)
            .load(url)
            .transform(CenterCrop(), RoundedCorners(24))
            .into(imageView)
}

@BindingAdapter("loadPosterWithBlur")
fun loadPosterWithBlur(imageView: ImageView, url: String?) {
    Glide.with(imageView.context)
            .load(url)
            .transform(BlurTransformation(imageView.context))
            .into(imageView)
}

@BindingAdapter("rating")
fun setRating(textView: TextView, rating: Double) {
    textView.text = rating.toString()
}
